### PR TITLE
 Contig db/api changes

### PR DESCRIPTION
**Pull request by yunfangjuan**
# Description
1. DB changes for loading contig related info 
2. working on API to get the contigs

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix a feature that knowingly causes existing functionality to not work as expected)

# Test and Reproduce
Outline the steps to test or reproduce the PR here.

1. 

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?

- [ ] Login Screen
- [ ] Single Upload Page
- [ ] Batch Upload Page
- [ ] All Projects Page
- [ ] Sample Details Page
- [ ] Report Page

Specific components 

* 

# Testing Script:

###Login Page
* Successfully login to the application and get taken to the project page

###Single Upload
* Using an AWS path that doesn't exist throws an error
* Succesfully uploaded files from a folder (use test host)
* Uploading a duplicate file throws an error
* Pipeline gets initiated successfully 

###Batch Upload
* Using an AWS path that doesn't exist throws an error
* Succesfully uploaded files from a folder (use test host)
* Navigate to batch upload settings page after upload 
* Host selection from previous page still selected 
* Uploading a duplicate file throws a warning
* Pipeline gets initiated successfully 

###All Project Page
* Switch between projects 
* Search for a sample
* Navigate through different pages of samples
* Select Upload new sample button
* Select sample to see generated report

###Sample Details Page
* Information is consistent from Project Page
* Notes are editable and save successfully 
* Can navigate to report page 

###Report Page
* Data loads succesfully 
* Collapse and Uncollapse button works
* Clicking on the taxonomy name gives you total reads for that taxon
* Category filter works
* Genus Search works
* Filters work as expected
* Disabling Filters works
* Filtering by column works 

# Checklist:

- [ ] I have run through the testing script to make sure current functionality is unchanged
- [ ] I have done relevant tests that prove my fix is effective or that my feature works
- [ ] I have spent time testing out edge cases for my feature
- [ ] I have updated the test script or pull request template if necessary
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: db/migrate/20181107023403_create_contigs.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class CreateContigs < ActiveRecord::Migration[5.1]
+  def change
+    add_column :pipeline_runs, :assembled, :integer, limit: 2
```
Comment (MarkAZhang) [inline db/migrate/20181107023403_create_contigs.rb]: I'm curious why we use type integer instead of type boolean for these flags.

**Comment by yunfangjuan** [comment]
i don't think rails supports boolean~

**Comment by jameshowardwang** [comment]
Maybe tinyint(1) then?

**Comment by tfrcarvalho** [comment]
Rails supports boolean: https://api.rubyonrails.org/classes/ActiveModel/Type/Boolean.html

**Comment by yunfangjuan** [comment]
To get list of taxids with legitimate contigs
http://localhost:3000/samples/979/contig_taxid_list 
To download contigs for a specific taxid
http://localhost:3000/samples/979/taxid_contigs?taxid=573

File: db/schema.rb [hunk 2]
```diff
@@ -32,9 +32,6 @@
     t.float "depth", limit: 24
     t.bigint "pipeline_run_id"
     t.string "drug_family"
-    t.integer "level"
-    t.float "drug_gene_coverage", limit: 24
```
Comment (jshoe) [inline db/schema.rb]: are these supposed to go away?

Comment (yunfangjuan) [reply db/schema.rb]: it's not in the migrations. i think it'sok.

**Pull request merged**